tests/sources/rest_api/test_rest_api_source.py

```diff
@@ -0,0 +1,116 @@
+import dlt
+import pytest
+from dlt.sources.rest_api.typing import RESTAPIConfig
+from dlt.sources.helpers.rest_client.paginators import SinglePagePaginator
+
+from dlt.sources.rest_api import rest_api_source
+from tests.utils import ALL_DESTINATIONS, assert_load_info, load_table_counts
+
+
+def _make_pipeline(destination_name: str):
+    return dlt.pipeline(
+        pipeline_name="rest_api",
+        destination=destination_name,
+        dataset_name="rest_api_data",
+        full_refresh=True,
+    )
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_rest_api_source(destination_name: str) -> None:
+    config: RESTAPIConfig = {
+        "client": {
+            "base_url": "https://pokeapi.co/api/v2/",
+        },
+        "resource_defaults": {
+            "endpoint": {
+                "params": {
+                    "limit": 1000,
+                },
+            }
+        },
+        "resources": [
+            {
+                "name": "pokemon_list",
+                "endpoint": "pokemon",
+            },
+            "berry",
+            "location",
+        ],
+    }
+    data = rest_api_source(config)
+    pipeline = _make_pipeline(destination_name)
+    load_info = pipeline.run(data)
+    print(load_info)
+    assert_load_info(load_info)
+    table_names = [t["name"] for t in pipeline.default_schema.data_tables()]
+    table_counts = load_table_counts(pipeline, *table_names)
+
+    assert table_counts.keys() == {"pokemon_list", "berry", "location"}
+
+    assert table_counts["pokemon_list"] == 1302
+    assert table_counts["berry"] == 64
+    assert table_counts["location"] == 1036
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
```

Comment:
you have 3 kinds of tests with regard to destinations
1. do not use destination
2. always use duckdb
3. runs on configured destinations

core has it's own way to do (3). look for tests/load/pipeline/test_pipelines how to you declare the pipeline and how you instantiate them. also put those in a separate file so we can easily select them. move them to `tests/load/sources/rest_api` then they will be picked up by existing CI automatically

Reply:
ie.
```
@pytest.mark.parametrize(
    "destination_config", destinations_configs(default_sql_configs=True), ids=lambda x: x.name
)
def test_skip_sync_schema_for_tables_without_columns(
    destination_config: DestinationTestConfiguration,
) -> None:
    # load data and then restore the pipeline and see if data is still there
    data = ["a", "b", "c"]

    @dlt.resource(name="data_table")
    def _data():
        for d in data:
            yield d

    p = destination_config.setup_pipeline("test_skip_sync_schema_for_tables", dev_mode=True)

```

Reply:
- [x] end-to-end tests using Pokemon API are extracted into `tests/load/sources/rest_api/test_rest_api_source.py`
- [x] end-to-end tests run on all destinations